File: maximum-xor-with-an-element-from-array.cc
#define ALL(x) (x).begin(), (x).end()
#define FOR(i, a, b) for (remove_cv<remove_reference<decltype(b)>::type>::type i = (a); i < (b); i++)
#define REP(i, n) FOR(i, 0, n)

class Solution {
public:
  vector<int> maximizeXor(vector<int>& nums, vector<vector<int>>& queries) {
    int n = nums.size(), m = queries.size();
    REP(i, m) queries[i].push_back(i);
    sort(

File: maximum-xor-with-an-element-from-array.rs
// Maximum XOR With an Element From Array
impl Solution {
    pub fn maximize_xor(mut nums: Vec<i32>, mut queries: Vec<Vec<i32>>) -> Vec<i32> {
        use std::usize::MAX;
        let n = nums.len();
        nums.sort_unstable();
        for (i, q) in queries.iter_mut().enumerate() { q.push(i as i32); }
        queries.sort_unstable_by_key(|q| q[1])